azure-ts-aks-keda/cluster.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016-2018, Pulumi Corporation.  All rights reserved.
+
+import * as azure from "@pulumi/azure";
+import * as azuread from "@pulumi/azuread";
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+
+// Arguments for an AKS cluster. We use almost all defaults for this examples, but the
+// interface could be extended with e.g. agent pool settings.
+export interface AksClusterArgs {
+    resourceGroup: azure.core.ResourceGroup;
+}
+
+export class AksCluster extends pulumi.ComponentResource {
+    public cluster: azure.containerservice.KubernetesCluster;
+    public provider: k8s.Provider;
+
+    constructor(name: string,
+                args: AksClusterArgs,
+                opts: pulumi.ComponentResourceOptions = {}) {
+        super("examples:keda:AksCluster", name, args, opts);
+
+        const config = new pulumi.Config();
+        const password = config.require("password");
```

Comment:
Should use `getSecret` here.
```suggestion
        const password = config.getSecret("password");
```

Reply:
Something like https://github.com/pulumi/examples/blob/733582dc6b10bffb9a5c6d502330af375fd13749/gcp-py-gke/__main__.py#L20 would avoid having to hardcode something here: https://github.com/pulumi/examples/pull/356/files#diff-35415c41b03f0c54711c098db019c53cR740

Reply:
Nice! Is there a similar thing to the ssh key? Should I get rid of the config values in this case? (not important for this example IMO)

Reply:
Yeah, we have other examples of config values, so might as well trim them here. I don't know of a similar approach for the SSH key.

Reply:
I got rid of configs, SSH key is also auto-generated now